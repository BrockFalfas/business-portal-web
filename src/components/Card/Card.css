.Card {
  background-color: red;
  padding: 0.8rem 1.3rem;
}

.Card-content {
  display: flex;
  justify-content: space-between;
  margin: 1.2rem 0 1rem;
}

.Card-text {
  text-align: right;
}

.Card-title {
  font-size: 1.5rem;
  font-weight: bold;
  margin: 0 0 0.6rem;
}
.Card-icon {
  font-size: 2.5rem;
}
.Card-description {
}
.Card-actions {
  text-align: right;
  height: 2rem;
}

.Card--white {
  background-color: white;
  color: #80d134;
}
.Card--white .Card-description {
  color: black;
}

.Card--black {
  background-color: black;
  color: #80d134;
}
.Card--black .Card-description {
  color: white;
}

.Card--blue {
  background-color: #4c92e0;
  color: white;
}
.Card--blue .Card-description {
  color: white;
}

.Card--green {
  background-color: #aadc74;
  color: white;
}
.Card--green .Card-description {
  color: black;
}
