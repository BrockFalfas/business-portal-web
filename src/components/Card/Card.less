@import "../../styles/variables.less";

.card-theme(@bg, @accent, @text) {
  & {
    background-color: @bg;
    color: @text;

    .Card-title,
    .Card-icon {
      color: @accent;
    }

    .Card-actions {
      .ant-btn {
        border-color: @accent;
        color: @accent;

        &:hover {
          background-color: @accent !important;
          color: @bg;
        }
      }
    }
  }
}

.Card {
  padding: 0.8rem 1.3rem;

  .Card-content {
    display: flex;
    justify-content: space-between;
    margin: 1.2rem 0 1rem;
  }

  .Card-text {
    text-align: right;
  }

  .Card-title {
    font-size: 1.5rem;
    font-weight: bold;
    margin: 0 0 0.6rem;
  }
  .Card-icon {
    font-size: 2.5rem;
  }
  .Card-description {
  }
  .Card-actions {
    text-align: right;
    height: 2rem;
  }
}

.Card--black {
  .card-theme(@color-black, @color-green-neon, @color-white);
}
.Card--blue {
  .card-theme(@color-blue, @color-white, @color-white);
}
.Card--green {
  .card-theme(@color-green, @color-white, @color-black);
}
