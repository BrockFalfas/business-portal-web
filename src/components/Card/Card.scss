@import "styles/variables";

@mixin card-theme($bg, $accent, $text) {
  $calcBg: map-get($namedColors, $bg);
  $calcAccent: map-get($namedColors, $accent);
  $calcText: map-get($namedColors, $text);

  & {
    background-color: $calcBg;
    color: $calcText;

    .Card-title,
    .Card-icon {
      color: $calcAccent;
    }

    .Card-actions {
      .ant-btn {
        border-color: $calcAccent;
        color: $calcAccent;

        &:hover {
          background-color: $calcAccent !important;
          color: $calcBg;
        }
      }
    }
  }
}

.Card {
  padding: 0.8rem 1.3rem;

  .Card-content {
    display: flex;
    justify-content: space-between;
    margin: 1.2rem 0 1rem;
  }

  .Card-text {
    text-align: right;
  }

  .Card-title {
    font-size: 1.5rem;
    font-weight: bold;
    margin: 0 0 0.6rem;
  }
  .Card-icon {
    font-size: 2.5rem;
  }
  .Card-description {
  }
  .Card-actions {
    text-align: right;
    height: 2rem;
  }
}

.Card--black {
  @include card-theme("black", "green-neon", "white");
}
.Card--blue {
  @include card-theme("blue", "white", "white");
}
.Card--green {
  @include card-theme("green", "white", "black");
}
