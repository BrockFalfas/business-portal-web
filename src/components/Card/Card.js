import React from 'react';
import PropTypes from 'prop-types';

import Box from '~components/Box';

import './Card.scss';

export class Card extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    color: PropTypes.oneOf(['black', 'blue', 'green']).isRequired,
    description: PropTypes.string,
    icon: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    rounded: PropTypes.bool,
    title: PropTypes.string,
  };

  render() {
    const { children, icon, title, description, rounded, color } = this.props;

    const Container = rounded ? Box : 'div';
    return (
      <Container className={`Card Card--${color}`}>
        <div className="Card-content">
          <div className="Card-icon">{icon}</div>
          <div className="Card-text">
            <div className="Card-title">{title}</div>
            <div className="Card-description">{description}</div>
          </div>
        </div>
        <div className="Card-actions">{children}</div>
      </Container>
    );
  }
}

export default Card;
