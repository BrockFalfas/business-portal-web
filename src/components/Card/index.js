import React from 'react';

import Box from '~components/Box';

import './Card.css';

export default class Card extends React.Component {
  static defaultProps = {
    color: 'white',
  };

  render() {
    const { icon, title, description, actions, rounded, color } = this.props;

    const Container = rounded ? Box : 'div';
    return (
      <Container className={`Card Card--${color}`}>
        <div className="Card-content">
          <div className="Card-icon">{icon}</div>
          <div className="Card-text">
            <div className="Card-title">{title}</div>
            <div className="Card-description">{description}</div>
          </div>
        </div>
        <div className="Card-actions">{actions}</div>
      </Container>
    );
  }
}
