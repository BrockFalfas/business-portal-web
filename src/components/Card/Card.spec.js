import React from 'react';
import { shallow } from 'enzyme';

import { Card } from './Card';

const initCard = overrides => {
  return shallow(<Card color="blue" {...overrides} />);
};

describe('component: <Card/>', () => {
  it('should render without crashing', () => {
    const wrapper = initCard();
    expect(wrapper).toBeTruthy();
  });

  it('should render children properly', () => {
    const fakeChild = <div>fake child</div>;
    const wrapper = initCard({ children: fakeChild });
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow setting different colors', () => {
    const allowedColors = ['black', 'blue', 'green'];
    allowedColors.forEach(color => {
      const wrapper = initCard({ color });
      expect(wrapper.prop('className')).toContain(`Card--${color}`);
    });
  });

  it('should render title, description and icon properly', () => {
    const wrapper = initCard({
      description: 'fake description',
      icon: <div>fake icon</div>,
      title: 'fake title',
    });
    expect(wrapper).toMatchSnapshot();
  });

  describe('container', () => {
    it('should use div element by default', () => {
      const wrapper = initCard({ rounded: false });
      expect(wrapper.is('div')).toBe(true);
    });

    it('should use Box component for rounded corners', () => {
      const wrapper = initCard({ rounded: true });
      expect(wrapper.is('Box')).toBe(true);
    });
  });
});
