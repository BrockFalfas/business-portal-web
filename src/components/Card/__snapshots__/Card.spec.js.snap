// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Card/> should render children properly 1`] = `
<div
  className="Card Card--blue"
>
  <div
    className="Card-content"
  >
    <div
      className="Card-icon"
    />
    <div
      className="Card-text"
    >
      <div
        className="Card-title"
      />
      <div
        className="Card-description"
      />
    </div>
  </div>
  <div
    className="Card-actions"
  >
    <div>
      fake child
    </div>
  </div>
</div>
`;

exports[`component: <Card/> should render title, description and icon properly 1`] = `
<div
  className="Card Card--blue"
>
  <div
    className="Card-content"
  >
    <div
      className="Card-icon"
    >
      <div>
        fake icon
      </div>
    </div>
    <div
      className="Card-text"
    >
      <div
        className="Card-title"
      >
        fake title
      </div>
      <div
        className="Card-description"
      >
        fake description
      </div>
    </div>
  </div>
  <div
    className="Card-actions"
  />
</div>
`;
